src/psyclone/transformations.py

```diff
@@ -2925,13 +2924,16 @@ def validate(self, node, options=None):
                     f" supported and kernel '{node.name}' is of type "
                     f"'{type(node).__name__}'")
 
-            # Get the PSyIR routine from the associated kernel
+            # Get the PSyIR routine from the associated kernel If there is an
+            # exception (this could mean that there is no associated tree
+            # or that the frontend failed to convert it into PSyIR) reraise it
+            # as a TransformationError
             try:
                 kernel_schedule = node.get_kernel_schedule()
-            except GenerationError as error:
+            except Exception as error:
```

Comment:
Is this form OK (was it a bare `except:` that you were saying we should avoid using the other day)?

Reply:
Yes, the bare `except:` includes things like keyboard or OS interrupts and should not be used. But these do not subclass Exception, so Exception is safe to use.